internal/testutils/jimmtest/mocks/jimm_role_mock.go

```diff
@@ -0,0 +1,72 @@
+// Copyright 2024 Canonical.
+
+package mocks
+
+import (
+	"context"
+
+	"github.com/canonical/jimm/v3/internal/common/pagination"
+	"github.com/canonical/jimm/v3/internal/dbmodel"
+	"github.com/canonical/jimm/v3/internal/errors"
+	"github.com/canonical/jimm/v3/internal/openfga"
+)
+
+// RoleManager is an implementation of the jujuapi.RoleManager interface.
+type RoleManager struct {
```

Comment:
We need a discussion on mocking stratgegy, I'd even put mock in the name... Uhm I want us to use gomock let's discuss in standup.